data/mesongenbuildreq.py

```diff
@@ -0,0 +1,16 @@
+import subprocess
+import json
+import sys
+deps_json = json.loads(subprocess.run([sys.argv[0], "introspect", "--dependencies", "meson.build"], capture_output=True).stdout)
+deps = dict(zip([x['name'] for x in deps_json],[x['version'] for x in deps_json]))
+deps.pop('', None)
+for lib, versions in deps.items() :
```

Comment:
Would be good to sort here to ensure deterministic output for reproducible builds.